app/src/main/java/org/mozilla/fenix/tabstray/browser/InactiveTabViewHolder.kt

```diff
@@ -5,165 +5,115 @@
-                showConfirmationSnackbar()
-            }
-        }
-
-        private fun showConfirmationSnackbar() {
-            val context = binding.root.context
-            val view = binding.root
-            val text = context.getString(R.string.inactive_tabs_auto_close_message_snackbar)
-            val snackbar = FenixSnackbar.make(
-                view = view,
-                duration = FenixSnackbar.LENGTH_SHORT,
-                isDisplayedWithBrowserToolbar = true
-            ).setText(text)
-            snackbar.view.elevation = TabsTrayFragment.ELEVATION
-            snackbar.show()
-        }
-
-        companion object {
-            const val LAYOUT_ID = R.layout.inactive_tabs_auto_close
         }
-    }
-
-    /**
-     * A RecyclerView ViewHolder implementation for an inactive tab view.
-     *
-     * @param itemView the inactive tab [View].
-     * @param featureName [String] representing the name of the feature displaying tabs. Used in telemetry reporting.
-     */
-    class TabViewHolder(
-        itemView: View,
-        private val delegate: TabsTray.Delegate,
-        private val featureName: String
-    ) : InactiveTabViewHolder(itemView) {
-
-        private val binding = InactiveTabListItemBinding.bind(itemView)
-
-        fun bind(tab: TabSessionState) {
-            val components = itemView.context.components
-            val title = tab.content.title.ifEmpty { tab.content.url.take(MAX_URI_LENGTH) }
-            val url = tab.content.url.toShortUrl(components.publicSuffixList).take(MAX_URI_LENGTH)
 
-            itemView.setOnClickListener {
-                TabsTrayMetrics.openInactiveTab.add()
-                delegate.onTabSelected(tab, featureName)
-            }
-
-            binding.siteListItem.apply {
-                components.core.icons.loadIntoView(iconView, tab.content.url)
-                setText(title, url)
-                setSecondaryButton(
-                    R.drawable.mozac_ic_close,
-                    R.string.content_description_close_button
-                ) {
+        if (listItems.isNotEmpty()) {
+            InactiveTabsList(
+                inactiveTabs = listItems,
+                expanded = expanded,
+                showAutoCloseDialog = showAutoClosePrompt,
+                onHeaderClick = { inactiveTabsInteractor.onHeaderClicked(!expanded) },
+                onDeleteAllButtonClick = tabsTrayInteractor::onDeleteInactiveTabs,
+                onAutoCloseDismissClick = {
+                    inactiveTabsInteractor.onCloseClicked()
+                    showAutoClosePrompt = !showAutoClosePrompt
+                },
+                onEnableAutoCloseClick = {
+                    inactiveTabsInteractor.onEnabledAutoCloseClicked()
+                    showAutoClosePrompt = !showAutoClosePrompt
+                    showConfirmationSnackbar()
+                },
+                onTabClick = {
+                    TabsTrayMetrics.openInactiveTab.add()
```

Comment:
For all the metrics here, can we move these into the controller function that is called through onTabSelected and can you add a test in the existing controller function?